operator/pkg/central/extensions/reconcile_admin_password.go

```diff
@@ -116,6 +119,12 @@ func (r *reconcileAdminPasswordExtensionRun) updateStatus(status *platform.Centr
 	if r.infoUpdate == status.Central.AdminPassword.Info {
 		return false
 	}
+
+	if r.autoGenerated || r.hasCredentials {
+		status.Central.AdminPassword.SecretReference = defaultPasswordSecretName
+	} else {
+		status.Central.AdminPassword.SecretReference = r.passwordSecretName
+	}
```

Comment:
Shouldn't this code also influence the return value like in the `if` block right above?

Reply:
I added `SecretReference` check so it effects which bool to return as well